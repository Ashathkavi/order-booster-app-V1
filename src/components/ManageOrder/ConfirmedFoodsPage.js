import React, {useEffect} from 'react'
import {connect} from 'react-redux'
import {startEditOrder} from '../../actions/orders'
import selectedOrders from '../../selectors/orders'
import {setStatusFilter} from '../../actions/orderFilters'
import categoryList from '../../fixtures/FoodCategoryList'
import moment from 'moment'


export const ConfirmedFoodsPage = (props) => {
    console.log(props.orders, 'props.orders')


    let soups=[] , apetizer =[], chopsueys=[], riceAndCurry=[], friedRice=[], briyani=[]
    let noodles=[] , spaghetti =[], pasta=[], kothu=[], vegetableDishes=[], eggDishes=[]
    let fishDishes=[] , chickenDishes =[], prawnDishes=[], cuttleFishDishes=[], crabDishes=[], desserts=[]
    let milkShakes=[] , freshFruitJuice =[], softDrink=[], sandwiches=[], specialMenue=[]
    
    props.orders.map((order)=>order.foods.map((food)=>{
        //console.log(food.food.category, 'food.food.category')
        if(food.food.category === 'soups'){
            soups.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Appetizer'){
            apetizer.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Chopsueys'){
            chopsueys.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Rice and Curry'){
            riceAndCurry.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Fried Rice'){
            friedRice.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Briyani'){
            briyani.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Noodles'){
            noodles.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Spaghetti'){
            spaghetti.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Pasta'){
            pasta.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Kothu'){
            kothu.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Vegetable Dishes'){
            vegetableDishes.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Egg Dishes'){
            eggDishes.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Fish Dishes'){
            fishDishes.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Chicken Dishes'){
            chickenDishes.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Prawn Dishes'){
            prawnDishes.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'CuttleFish Dishes'){
            cuttleFishDishes.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Crab Dishes'){
            crabDishes.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Desserts'){
            desserts.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'MilkShakes'){
            milkShakes.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Fresh Fruit Juices'){
            freshFruitJuice.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Soft Drink'){
            softDrink.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Sandwiches'){
            sandwiches.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Special Menue'){
            specialMenue.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }
    }))

    console.log(spaghetti, 'spaghetti')

    const categoryList = [
        soups , apetizer , chopsueys, riceAndCurry, friedRice, briyani,
        noodles , spaghetti , pasta, kothu, vegetableDishes, eggDishes,
        fishDishes , chickenDishes , prawnDishes, cuttleFishDishes, crabDishes, desserts,
        milkShakes , freshFruitJuice , softDrink, sandwiches, specialMenue
    ]

    const onChangePreparedStatus = (id, food, orderedFoods) => {
        let foods = orderedFoods
        foods[food.index].prepared = true
        foods[food.index].preparedBy = props.auth.uid
        foods[food.index].preparedTime = moment().valueOf()
        let orderReady = true
        foods.map((food)=> {
            if(food.prepared === false){
                orderReady = false
            }
        })
        if(orderReady){
            props.startEditOrder(id ,{foods:foods, status:{status:'table', time:moment().valueOf()}})
        }else{
            props.startEditOrder(id ,{foods:foods})
        }
        console.log(foods, 'Foods', id)
        
    }


    useEffect(()=>{
        if(props.filters.status !== 'kitchen'){
            props.setStatusFilter('kitchen')
        }
    }, ()=>{

    })


    
    return(
        <div>
        {
            categoryList.map((category)=>{
                
                if(category.length !== 0){
                    console.log(category, 'category')
                    return (
                        <div>
                            <h3>{category[0].food.category}</h3>
                            {
                                
                                category.map((food)=>{
                                    let orderedFoods = []
                                    props.orders.map((order)=>{
                                        if(order.id === food.orderId){
                                            orderedFoods = order.foods
                                            console.log(orderedFoods, 'orderedFoods')
                                        }
                                    })
                                    
                                    return (
                                        <div>
                                            <p>{food.food.name}:::
                                            {food.food.foodSize}:::
                                            {food.foodQuantity}:::
                                            {food.orderNo}:::</p>
                                            {food.prepared === false && <button onClick={()=>onChangePreparedStatus(food.orderId, food, orderedFoods)}>Mark it as Prepared</button>}

                                        </div>
                                    )
                                        
                                })
                            }
                        </div>
                    )
                }
            })
        }
        </div>
        
            
        
    )
}


const mapDispatchToProps = (dispatch) => ({
    startEditOrder: (id, order) => dispatch(startEditOrder(id, order)),
    setStatusFilter: (status) => dispatch(setStatusFilter(status))

})

const mapStateToProps = (state) => {
    return {        
        orders:selectedOrders(state.orders, state.orderFilters),
        filters:state.orderFilters,
        auth:state.auth
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(ConfirmedFoodsPage)