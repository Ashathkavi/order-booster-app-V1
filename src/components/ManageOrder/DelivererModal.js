import React, {useEffect, useState} from 'react'
import Modal from 'react-modal'
import CategoryList from '../../fixtures/FoodCategoryList'
import database from '../../firebase/firebase'


class Deliverer{
    constructor(name, uid){
        this.name = name
        this.uid = uid
    }
}

const useFetchDeliverer = () => {
    const [value, setValue] = useState([]);
    useEffect(()=>{
        const deliverers = []
        database.ref(`users`).once('value')
            .then((snap)=>{
                snap.forEach((childSnapshot)=>{
                    if(childSnapshot.val().role === 'deliverer'){
                        const deliverer = new Deliverer(childSnapshot.val().name, childSnapshot.key)
                        deliverers.push(deliverer)                        
                    }
                })
                setValue(deliverers)                    
            })
            .catch((error)=>console.log('User gaining request failed :', error))
        
    },[])
    return [value, setValue]
}



const DelivererModal = (props) => {

    const [deliverers, setDeliverers] = useFetchDeliverer()

    

    const arrayBuilder = (array, devideBy) => {
        let DelivererListV1 = []
        let DelivererListV2 = []
        let i = 0
        array.map((item)=>{
            DelivererListV1.push(item)
            i = i +1

            if(i===devideBy){
                DelivererListV2.push(DelivererListV1)
                i=0
                DelivererListV1=[]
            }

            if(array.indexOf(item) === array.length - 1){
                DelivererListV2.push(DelivererListV1)
                console.log()
            }

        }) 
        return DelivererListV2

    }
    const modifiedDelList = arrayBuilder(deliverers, 2)
    

    return(
        <Modal
            isOpen={props.modalIsOpen}
            contentLabel = "Please Select a Deliverer"
            onRequestClose = {props.handleCloseModal}
        >
            <h3 className="modal__header" >Please Select a Deliverer</h3>
            <div>
            
                {
                    modifiedDelList.map((internalDelList)=>{
                        console.log(modifiedDelList)
                        return(
                            <div className="modal__content">
                                {
                                    internalDelList.map((deliverer)=> (
                                        <div className="modal__content__item__deliverer">
                                            <button className="button button--selectFood" onClick={() =>{                
                                                props.onDelivery(deliverer.uid)
                                                props.handleCloseModal()
                                            }}>{deliverer.name}</button>
                                        </div>
                                    ))
                                    
                                }
                                {
                                    internalDelList[1] === undefined ? <div className="modal__content__item__deliverer"></div> : null
                                    
                                }
                            </div>
                        )
                    })
                }
            </div>
            <div className="modal__buttons">
                    
                <button className="button button--closeModal" onClick={()=>props.handleCloseModal()}>Close</button>
                <button className="button button--selectNone" onClick={()=>{
                    props.onDelivery('') 
                    props.handleCloseModal()
                }}>Select None</button>
            </div>
            
        </Modal>
)}

export default DelivererModal;


/*

*/