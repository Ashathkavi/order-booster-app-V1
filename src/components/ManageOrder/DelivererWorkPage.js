import React, {useEffect, useState} from 'react'
import { connect } from 'react-redux'
import moment from 'moment'
import numeral from 'numeral'
import {startEditOrder} from '../../actions/orders'
import {setDeliverMethFilter, setStartDate} from '../../actions/orderFilters'
import OrderViewModal from './OrderViewModal'
import {onStartSetOrders} from '../../actions/orders'
import selectedOrders from '../../selectors/orders'



export const DelivererWorkPage = ({
    orders,
    setDeliverMethFilter,
    onStartSetOrders,
    setStartDate,
    startEditOrder

    }) => {
    //console.log(props)

    useEffect(()=>{
        setDeliverMethFilter('delivery')
        setStartDate(moment().subtract(1,'month').startOf('month').valueOf())
        const g = onStartSetOrders()
        return g
    },[])
    const [orderViewIsOpen, setOrderViewIsOpen ] = useState(false)

    const now = moment()

    const onChangeStatus = () => {
        startEditOrder(id ,{status:{status:'recieved', time:now.valueOf()}})
    }


    return(
        <div className='content-container'>
        
            <div className="list-body--deliverer">
            {
                orders.map((order)=>(
                    <div className="list-item list-item--deliverer">
                        
                        <h2 className="list-item__title list-item__title--deliverer">{order.count}  &nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp; {numeral(order.amount).format('$0,0.00')}</h2>
                        <p><b>Customer Name: </b> {order.customerName}</p>
                        <p><b>Phone Number: </b>{order.phoneNumber}</p>
                        <p><b>Address: </b>{order.address}</p>    
                        {order.status.status === 'on delivery' && <button onClick={onChangeStatus}>Mark it as Delivered</button>}
                        <p >{moment(order.orderEndTime).format('MMMM Do, YYYY   h:mm a') }</p>
                        <button className="button button--buttonMainBoard" onClick={()=>setOrderViewIsOpen(true)}>See Order Details</button> &nbsp;&nbsp;&nbsp;
                    
                        <OrderViewModal 
                            modalIsOpen={orderViewIsOpen} 
                            handleCloseModal={()=>setOrderViewIsOpen(false)} 
                            {...order}                
                            startEditOrder={startEditOrder}
                            from={'delivererList'}
                        />
                                
                    </div>
                ))
            }
            </div>
        </div>
        
    )

}



const mapDispatchToProps = (dispatch) => ({
    startEditOrder: (id, order) => dispatch(startEditOrder(id, order)),
    setDeliverMethFilter:(deliverMeth) => dispatch(setDeliverMethFilter(deliverMeth)),
    setStartDate:(startDate)=>dispatch(setStartDate(startDate)),
    onStartSetOrders: () => dispatch(onStartSetOrders())

})

const mapStateToProps = (state) => {
    return {        
        orders:selectedOrders(state.orders, state.orderFilters),
        filters:state.orderFilters,
        auth:state.auth
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(DelivererWorkPage)