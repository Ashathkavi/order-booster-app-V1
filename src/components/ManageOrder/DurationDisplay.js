import React, {useState,useEffect} from 'react'
import moment from 'moment'


const DurationDisplay = ({orderEndTime}) => {
    const [time, setTime] = useState(moment().valueOf())

    useEffect(() => {
        setTimeout(() => setTime(moment().valueOf()), 60000)
    }, [time])

    let duration = moment(orderEndTime).diff(moment())
    let durationPlusTen = moment(orderEndTime).subtract(10,'minutes').diff(moment())
    if(duration < 0){
        if(moment.utc(duration*-1).format('DD')==='01'){
            return <span style={{color: "red"}}>{moment.utc(duration*-1).format('HH ')}h&nbsp;&nbsp;{moment.utc(duration*-1).format('mm ')}min</span>
        }else{
            return <span style={{color: "red"}}>
                        {moment.utc(duration*-1).format('DD ')}d&nbsp;&nbsp;
                        {moment.utc(duration*-1).format('HH ')}h&nbsp;&nbsp;
                        {moment.utc(duration*-1).format('mm ')}min
                    </span>
        }
        //console.log(moment.utc(duration*-1).format('DD'))
    }else if(durationPlusTen < 0){
        return <span style={{color: "orange"}}>{moment.utc(duration).format('mm ') }min</span>
    }else{
        if(moment.utc(duration).format('HH')==='00'){
            return <span >{moment.utc(duration).format('mm ')}min</span>
        }else{
            return <span >{moment.utc(duration).format('HH ')}h&nbsp;&nbsp;{moment.utc(duration).format('mm ')}min</span>
        }
        
    }
    

} 

export default DurationDisplay