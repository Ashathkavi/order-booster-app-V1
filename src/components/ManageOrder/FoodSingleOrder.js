import React from 'react'
import {Link} from 'react-router-dom'

const FoodSingleOrder = (props) => (
    <tr>
        <td>{props.singleTypeOrder.food.name}</td>
        <td>{props.singleTypeOrder.foodQuantity}</td>
        <td>{props.singleTypeOrder.prepared ? "true" : "false"}</td>
        {!!props.onRemoveSingleOrder && <td><button type='button' onClick={()=>props.onRemoveSingleOrder(props.iterable)}>Remove</button></td>}
            
    </tr>
)

export default FoodSingleOrder