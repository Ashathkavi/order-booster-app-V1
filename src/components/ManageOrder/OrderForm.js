import React from 'react'
import moment from 'moment'
import DatePicker from 'react-datepicker'
import {connect} from 'react-redux'
import OrderModal from './OrderModal'
import FoodSingleOrder from './FoodSingleOrder'



const now = moment()
//console.log(now.format('MMM do, YYYY'))
const defaultOrderDuration = '20'

class StatusTimer{
    constructor(status, time){
        this.status = status
        this.time = time
    }
}


export class OrderForm extends React.Component{
     
    constructor(props){
        super(props)
        this.state = {
            isModalOpen:false,

            customerName:props.order ? props.order.customerName : '',            
            phoneNumber:props.order ? props.order.phoneNumber : 0,
            address:props.order ? props.order.address : '',
            orderEndTime:props.order ? props.order.orderEndTime : moment().add(20, 'minutes').valueOf(),
            duration:defaultOrderDuration,
            description:props.order ? props.order.description : '', 

            status:props.order ? props.order.status.status : 'confirmed',
            statusTime: props.order ? props.order.status.time : moment().valueOf(),

            kotStatus:props.order ? props.order.kotStatus.status : 'not',
            kotStatusTime: props.order ? props.order.kotStatus.time : moment().valueOf(),

            billStatus:props.order ? props.order.billStatus.status : 'not',
            billStatusTime: props.order ? props.order.billStatus.time : moment().valueOf(), 

            foods:props.order ? props.order.foods : [],

            //amount:props.order ? (props.order.amount / 100).toString() : '', 
            createdAt:props.order ?  moment(props.order.createdAt).valueOf(): moment().valueOf(),
            error:''
        }
    }

        //handling customer Name
        onCustomerNameChange = (e) => {
            e.persist()
            this.setState(()=>({customerName:e.target.value}))
        }  

        //handling customer Phonenumber
        onPhoneNumberChange = (e) => {
            const phoneNumber = e.target.value
            //if(phoneNumber.match(/^\d{1,}$/gm)){
                this.setState(()=>({phoneNumber: e.target.value}))
            //} 
        }

        //handling customer Address
        onAddressChange = (e) => {
            e.persist()
            this.setState(()=>({address:e.target.value}))       
        }

        //handling order duration
        onDurationChange = (e) => {
            const orderDuration = parseInt(e.target.value)
            this.setState(()=>({
                orderEndTime:moment(this.state.createdAt).add(orderDuration,'minutes').valueOf(),
                duration:e.target.value
            })) 
        }

        //handling Order Description
        onDescriptionChange = (e) => { 
            e.persist()
            this.setState(()=>({description:e.target.value}))
        }

        //handling order Status
        onStatusChange = (e) => {
            e.persist()
            this.setState(()=>({
                status:e.target.value,
                statusTime:moment().valueOf()
            }))       
        }

        //handling kot Status
        onKotStatusChange = (e) => {
            e.persist()
            this.setState(()=>({
                kotStatus:e.target.value,
                kotStatusTime:moment().valueOf()
            }))       
        }

        //handling bill Status
        onBillStatusChange = (e) => {
            e.persist()
            this.setState(()=>({
                billStatus:e.target.value,
                billStatusTime:moment().valueOf()
            }))       
        }

        //handling create date
        onDateChange = (date) => {
            //e.persist()
            const date_value = moment(date).valueOf()
            //console.log(date_value,'date')
            this.setState(()=>({
                createdAt:date_value,
                orderEndTime:moment(date_value).add(defaultOrderDuration, 'minutes').valueOf()
            }))
            //this.setState(()=>({billStatus:e.target.value}))       
        }

        //handling Food Modal  
        onVisibleChange = () => {
            this.setState((prevState)=>({isModalOpen: !prevState.isModalOpen}))
        }

        //handle single Order 
        onSingleOrderChange = (singleOrders) => {
            //console.log('singleOrders', singleOrders)
            this.setState((prevState)=>{
                //console.log('prevState.foods', prevState.foods)
                return {foods: [...singleOrders]}
            })
        }

        //removing selected single order  
        onRemoveSingleOrder = (arrayPosition) => {

            this.setState((prevState)=>{
                if(prevState.foods.length !== 1){
                    if(arrayPosition === 0){
                        return ({foods:prevState.foods.slice(1)})
                    }
                    //console.log('fdffffffffffff', arrayPosition)
                    let foods = prevState.foods
                    foods.splice(arrayPosition, 1)
                    return ({foods:foods})
                }            
                return ({foods:[]})
            })
        }

        //handle total bill amount calculation  //................................not tested
        onBillAmountCalculation = () => {
            let billAmount = 0
            this.state.foods.map((food)=>billAmount=(food.food.amount*food.foodQuantity )+ billAmount)
            //console.log(billAmount)
            return billAmount
        }

    onSubmit = (e) => {
        e.preventDefault()
        if(!this.state.customerName || !this.state.phoneNumber || !this.state.address ||
            !this.state.orderEndTime || this.state.foods === []){
            this.setState(()=>({
                error:'CUSTOMER NAME, PHONENUMBER, ADDRESS , ORDERDURATIONS are ORDER PLACINGS are mandatoroy to fill'
            }))
        }else{//new StatusTimer(this.state.billStatus, this.state.billStatusTime)
            this.setState(()=>({error:''}))
            this.props.onSubmit({
                address:this.state.address,
                amount:this.onBillAmountCalculation(),
                createdAt:this.state.createdAt,
                billStatus:{status:this.state.billStatus, time:this.state.billStatusTime} ,
                customerName:this.state.customerName,
                description:this.state.description,
                foods:this.state.foods,
                kotStatus:{status:this.state.kotStatus, time:this.state.kotStatusTime} ,
                orderEndTime:this.state.orderEndTime,
                phoneNumber:this.state.phoneNumber,
                status:{status:this.state.status, time:this.state.statusTime} 
            })
        }
    }
    





    render(){
        let iterable = -1
        //console.log('this.props.foods', this.props.foods)
        return(
            <div>
                {this.state.error && <p>{this.state.error}</p>}
                <form onSubmit={this.onSubmit}>
                    <button type='button' onClick={this.onVisibleChange}>Add Food</button>

                    <OrderModal 
                        onVisibleChange={this.onVisibleChange} 
                        isModalOpen = {this.state.isModalOpen}
                        onSingleOrderChange={this.onSingleOrderChange}
                        singleTypeOrders = {this.state.foods}
                        
                    />
                    {/*Adding Customer Name*/}
                    <input
                        type="text"
                        placeholder="Customer Name"
                        value={this.state.customerName}
                        onChange={this.onCustomerNameChange}                        

                    />

                    {/*Adding Customer Name*/}
                    <input
                        type="number"
                        placeholder="Phone Number"
                        value={this.state.phoneNumber}
                        onChange={this.onPhoneNumberChange }                        

                    />

                    {/*Adding Customer Address*/}
                    <input
                        type="text"
                        placeholder="Customer Address"
                        value={this.state.address}
                        onChange={this.onAddressChange }                        

                    />

                    {/*Adding Description*/}
                    <textarea 
                        placeholder="Add a discription for your order (optional)"
                        autoFocus 
                        value={this.state.description}
                        onChange={this.onDescriptionChange}>
                    </textarea>

                    {
                        
                        this.state.foods.map((food)=>{
                            iterable = iterable + 1
                            return <FoodSingleOrder 
                                    key = {iterable}
                                    iterable = {iterable}
                                    singleTypeOrder = {food}
                                    onRemoveSingleOrder = {this.onRemoveSingleOrder}
                                />
                        })
                        
                    }
                    {
                        <div><h3>:::::::::::::{this.onBillAmountCalculation().toString()}:::::::::::::</h3></div>
                        
                    }

                    {/*Adding Status*/}                    
                    <select value={this.state.status} onChange={this.onStatusChange} >
                        <option value ='confirmed'>Confirmed</option>
                        <option value ='cancelled'>Cancelled</option>
                        <option value ='kitchen'>Kitchen</option>
                        <option value ='table'>Table</option>
                        <option value ='on delivery'>On Delivery</option>
                        <option value ='recieved'>Recieved</option>
                    </select>

                    {/*Adding KOT status*/}  
                    <select value={this.state.kotStatus} onChange={this.onKotStatusChange} disabled={!this.props.order}>
                        <option value ='not'>Not Passed</option>
                        <option value ='passed'>Passed</option>
                    </select>

                    {/*Adding Bill Status*/}  
                    <select value={this.state.billStatus} onChange={this.onBillStatusChange} disabled={!this.props.order}>
                        <option value ='not'>Not Printed</option>
                        <option value ='printed'>Printed</option>
                    </select>                    

                    {/*Adding Order Duration*/}
                    <select value={this.state.duration} onChange={this.onDurationChange} >
                        <option value ='20'>20 minutes</option>
                        <option value ='30'>30 minutes</option>
                        <option value ='40'>40 minutes</option>
                        <option value ='50'>50 minutes</option>
                        <option value ='60'>1 hour</option>
                        <option value ='70'>1 hour 10 minutes</option>
                        <option value ='80'>1 hour 20 minute</option>
                        <option value ='90'>1 and 1\2 hour </option>
                    </select>

                    {/*Adding Order Creation Date*/}
                    <DatePicker
                        selected={moment(this.state.createdAt).toDate()}
                        onChange={this.onDateChange}
                        showTimeSelect
                        timeFormat="HH:mm"
                        dateFormat="MMMM d, yyyy h:mm aa"
                    />
                    <button>Save Order</button>

                </form>

                <div>
                    <p>customerName: {this.state.customerName}</p>
                    <p>phoneNumber: {this.state.phoneNumber}</p>
                    <p>address: {this.state.address}</p>
                    <p>description: {this.state.description}</p>
                    <p>orderEndTime:  {moment(this.state.orderEndTime).toLocaleString() }</p>
                    <p>status: {this.state.status}</p>
                    <p>kotSatus: {this.state.kotStatus}</p>
                    <p>billStatus: {this.state.billStatus}</p>
                    <p>createdAt: {moment(this.state.createdAt).toLocaleString()}</p>
                    <p>foods: {this.state.foods.toString()}</p>
                    
                    <p>isModaOpen: {this.state.isModalOpen.toString()}</p>

                </div>   
            </div>
            
        )
    
    }
}


const mapStateToProps = (state, props) => {
    console.log(state.foods)
    return {
        
        foods:state.foods
    }
}

export default connect(mapStateToProps)(OrderForm)