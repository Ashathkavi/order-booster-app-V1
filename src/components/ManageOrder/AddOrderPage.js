import React from 'react'
import {OrderForm} from './OrderForm'
import {startAddOrder} from '../../actions/orders'
import {connect} from 'react-redux'
import selectFood from '../../selectors/foods'



export const AddOrderPage = (props) => {

    
    const counter = (orders) =>{
        let count = orders.length + 1
        let i = true
        while(i === true){

            orders.map((order) => {
                i=false
                if(order.count  === count){
                    count += 1
                    i=true
                }
            })
            if(orders.length === 0){
                i = false
            }
        }
        return count
    } 

    const onSubmit = (order) => {
        props.startAddOrder({...order, count:counter(props.orders)})
        props.history.push('/order')
    }

    return(
        <div>
            <div className="page-header">
                <div className="content-container">
                    <h1 className="page-header__title">Add Order</h1>
                </div>
            </div>

            <div className="content-container">
                <OrderForm orders={props.orders} onSubmit={ onSubmit} foods={props.foods}/>
            </div>
            

        </div>
)}


const mapDispatchToProps = (dispatch) => ({
    startAddOrder: (order) => dispatch(startAddOrder(order))
})


const mapStateToProps = (state) => {
    console.log(state.orders)
    return {
        orders:state.orders,
        foods:selectFood(state.foods, state.foodFilters)
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(AddOrderPage)