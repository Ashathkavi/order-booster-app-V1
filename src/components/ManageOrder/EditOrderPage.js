import React from 'react'
import {connect} from 'react-redux'
import { editOrder, removeOrder} from '../../actions/orders'
import OrderForm from './OrderForm'

export const EditOrderPage = (props) => {
    
    const onSubmit = (order) => {
        props.editOrder(props.order.id ,order)
        props.history.push('/order')
    }

    const onRemove = () => {
        props.removeOrder(props.order.id)
        props.history.push('/order')
    }

    return(
        <div>
           <OrderForm order={props.order} onSubmit={onSubmit}/>
           <button onClick={onRemove}>Remove</button>
        </div>
)}


const mapDispatchToProps = (dispatch) => ({
    editOrder: (id, order) => dispatch(editOrder(id, order)),
    removeOrder: (id) => dispatch(removeOrder({id}))
})

const mapStateToProps = (state, props) => {
    //console.log(state.orders)
    return {
        
        order:state.orders.find((order)=> order.id === props.match.params.id)
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(EditOrderPage)