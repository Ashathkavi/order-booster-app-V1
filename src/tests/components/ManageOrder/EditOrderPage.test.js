import React from 'react'
import {shallow} from 'enzyme'
import {EditOrderPage} from '../../../components/ManageOrder/EditOrderPage'
import sampleOrders from '../../../fixtures/sampleOrders'
import OrderForm from '../../../components/ManageOrder/OrderForm'


const sample_orders = sampleOrders()

let editOrderSpy, historySpy, wrapper, startRemoveOrderSpy
 
beforeEach(()=>{
    editOrderSpy = jest.fn()
    startRemoveOrderSpy = jest.fn()
    historySpy = {push: jest.fn()}
    wrapper = shallow(<EditOrderPage 
        editOrder={editOrderSpy} 
        startRemoveOrder = {startRemoveOrderSpy} 
        history={historySpy}     
        order = {sample_orders[0]}   
    />)
})

test('should render EditOrderPage correctly', () => {
    expect(wrapper).toMatchSnapshot();
})

test('should handle onSubmit', () => {
    wrapper.find(OrderForm).prop('onSubmit')(sample_orders[1])
    expect(historySpy.push).toHaveBeenLastCalledWith('/order')
    expect(editOrderSpy).toHaveBeenLastCalledWith(sample_orders[0].id, sample_orders[1])
})

test('should handle onSubmit', () => {
    wrapper.find('button').simulate('click')
    expect(historySpy.push).toHaveBeenLastCalledWith('/order')
    expect(startRemoveOrderSpy).toHaveBeenLastCalledWith(sample_orders[0].id)
})