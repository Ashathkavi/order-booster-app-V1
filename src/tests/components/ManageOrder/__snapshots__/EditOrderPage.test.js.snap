// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditOrderPage correctly 1`] = `
<div>
  <Connect(OrderForm)
    onSubmit={[Function]}
    order={
      Object {
        "address": "Mamangam",
        "amount": 1810,
        "billStatus": Object {
          "status": "printed",
          "time": 300000,
        },
        "createdAt": 0,
        "customerName": "Ashath",
        "description": "less oil",
        "foods": Array [
          SingleOrder {
            "food": Object {
              "amount": 380,
              "category": "Fried Rice",
              "createdAt": 100001,
              "foodSize": "regular",
              "id": 2,
              "largeAvailability": true,
              "name": "Beef Fried Rice",
            },
            "foodQuantity": 2,
          },
          SingleOrder {
            "food": Object {
              "amount": 350,
              "category": "Fried Rice",
              "createdAt": 100000,
              "foodSize": "regular",
              "id": 1,
              "largeAvailability": true,
              "name": "Chicken Fried Rice",
            },
            "foodQuantity": 3,
          },
        ],
        "id": 1,
        "kotStatus": Object {
          "status": "passed",
          "time": 300000,
        },
        "orderEndTime": 1200000,
        "phoneNumber": "0765822562",
        "status": Object {
          "status": "confirmed",
          "time": 0,
        },
      }
    }
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
